src/buildApplication/buildVercelOutput.ts

```diff
@@ -44,9 +40,48 @@ async function generateProjectJsonFileIfNeeded(): Promise<void> {
 	}
 }
 
-async function runVercelBuild(): Promise<void> {
-	const npx = process.platform === 'win32' ? 'npx.cmd' : 'npx';
-	const vercelBuild = spawn(npx, ['vercel', 'build']);
+async function runVercelBuild(
+	pkgMng: 'pnpm' | 'yarn (berry)' | 'yarn (classic)' | 'npm'
+): Promise<void> {
+	cliLog(`Building project...`);
+
+	let cmd = '',
+		args = [];
+	const winCMD = process.platform === 'win32' ? '.cmd' : '';
+
+	if (pkgMng === 'yarn (classic)') {
+		cliLog(
+			`Installing vercel as dev dependencies with 'yarn add vercel -D'...`
+		);
+
+		const installVercel = spawn(`yarn${winCMD}`, ['add', 'vercel', '-D']);
+
+		installVercel.stdout.on('data', data => cliLog(`\n${data}`, true));
+		installVercel.stderr.on('data', data => cliError(`\n${data}`, false, true));
+
+		await new Promise((resolve, reject) => {
+			installVercel.on('close', code => {
+				if (code === 0) {
+					resolve(null);
+				} else {
+					reject();
+				}
+			});
+		});
+
+		cliLog(`Install completed`);
+
+		(cmd = `yarn${winCMD}`), (args = ['vercel', 'build']);
+	}
+
+	if (pkgMng === 'pnpm') (cmd = `pnpx${winCMD}`), (args = ['vercel', 'build']);
+	if (pkgMng === 'npm') (cmd = `npx${winCMD}`), (args = ['vercel', 'build']);
+	if (pkgMng === 'yarn (berry)')
+		(cmd = `yarn${winCMD}`), (args = ['dlx', 'vercel', 'build']);
+
+	cliLog(`Building project with '${cmd} ${args.join(' ')}'...`);
+
+	const vercelBuild = spawn(cmd, args);
```

Comment:
I feel like having `cmd` and `args` makes this much less readable and more complicated than needed (you than have to construct your command and args multiple times with different if statements)

what would you think of something along these lines instead?
![Screenshot 2023-04-05 at 09 37 51](https://user-images.githubusercontent.com/61631103/230027959-68a1a75c-97f7-4cb5-804f-e74c75ab8559.png)